### PR TITLE
Fix update sample version when releasing

### PR DESCRIPTION
**Pull request by iocanel**
Follow up fix for: #556

### CODE REVIEW

**Comment by geoand** [comment]
Did you run this to see that is make the change we want (to the `quarkus-langchain4j.version` property and not the `version` of the project itself)?

**Comment by jmartisk** [comment]
Huhm, how could I be so dumb to not notice..
Yeah, to update the property, you need to do
```
 ./mvnw org.codehaus.mojo:versions-maven-plugin:2.16.2:update-property -Dproperty=quarkus-langchain4j.version -DnewVersion=${{steps.metadata.outputs.current-version}} -DallowDowngrade=true -f samples/email-a-poem
```

.. for all samples

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -58,13 +58,13 @@ jobs:
...
+          ./mvnw org.codehaus.mojo:versions-maven-plugin:2.16.2:set -DnewVersion=${{steps.metadata.outputs.current-version}} -f samples/email-a-poem
+          ./mvnw org.codehaus.mojo:versions-maven-plugin:2.16.2:set -DnewVersion=${{steps.metadata.outputs.current-version}} -f samples/fraud-detection
+          ./mvnw org.codehaus.mojo:versions-maven-plugin:2.16.2:set -DnewVersion=${{steps.metadata.outputs.current-version}} -f samples/review-triage
+          ./mvnw org.codehaus.mojo:versions-maven-plugin:2.16.2:set -DnewVersion=${{steps.metadata.outputs.current-version}} -f samples/chatbot
+          ./mvnw org.codehaus.mojo:versions-maven-plugin:2.16.2:set -DnewVersion=${{steps.metadata.outputs.current-version}} -f samples/csv-chatbot
+          ./mvnw org.codehaus.mojo:versions-maven-plugin:2.16.2:set -DnewVersion=${{steps.metadata.outputs.current-version}} -f samples/cli-translator
+          ./mvnw org.codehaus.mojo:versions-maven-plugin:2.16.2:set -DnewVersion=${{steps.metadata.outputs.current-version}} -f samples/chatbot-easy-rag
```
Comment (gsmet) [inline .github/workflows/release.yml:67]: Probably a good idea to use some bash to automate it?

```
for module in samples/*; do echo $module; done
```

Make sure you define the shell as bash.

Comment (jmartisk) [reply .github/workflows/release.yml:67]: I thought about that too, but there's one sample (the jbang one) that isn't maven based so we'd have to exclude it, and that felt ugly enough for me to persuade me to go with hardcoding the list of modules. But of course we can do it.

Comment (gsmet) [reply .github/workflows/release.yml:67]: If you insist:

```
for pom in `find samples/ -name pom.xml`; do module=$(dirname "$pom"); echo $module; done
```

Comment (geoand) [reply .github/workflows/release.yml:67]: @iocanel do you want to update the PR as per Guillaume's suggestion?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -58,13 +58,9 @@ jobs:
...
-          ./mvnw org.codehaus.mojo:versions-maven-plugin:2.16.2:set -DnewVersion=${{steps.metadata.outputs.current-version}} -pl samples/fraud-detection
-          ./mvnw org.codehaus.mojo:versions-maven-plugin:2.16.2:set -DnewVersion=${{steps.metadata.outputs.current-version}} -pl samples/review-triage
-          ./mvnw org.codehaus.mojo:versions-maven-plugin:2.16.2:set -DnewVersion=${{steps.metadata.outputs.current-version}} -pl samples/chatbot
-          ./mvnw org.codehaus.mojo:versions-maven-plugin:2.16.2:set -DnewVersion=${{steps.metadata.outputs.current-version}} -pl samples/csv-chatbot
-          ./mvnw org.codehaus.mojo:versions-maven-plugin:2.16.2:set -DnewVersion=${{steps.metadata.outputs.current-version}} -pl samples/cli-translator
-          ./mvnw org.codehaus.mojo:versions-maven-plugin:2.16.2:set -DnewVersion=${{steps.metadata.outputs.current-version}} -pl samples/chatbot-easy-rag
+          for pom in `find samples/ -name pom.xml`; 
+            ./mvnw org.codehaus.mojo:versions-maven-plugin:2.16.2:set -DnewVersion=${{steps.metadata.outputs.current-version}}  -f $(dirname "$pom");
```
Comment (geoand) [inline .github/workflows/release.yml:62]: Shouldn't it be ` ./mvnw org.codehaus.mojo:versions-maven-plugin:2.16.2:update-property -Dproperty=quarkus-langchain4j.version  ...` ?

File: .github/workflows/release.yml (hunk 1 above)
Comment (geoand) [reply .github/workflows/release.yml]: I have added a comment below

File: .github/workflows/release.yml (hunk 2 above)
Comment (iocanel) [reply .github/workflows/release.yml]: Updated to use set-property.

**Review by geoand — APPROVED** [review]

**Pull request merged**